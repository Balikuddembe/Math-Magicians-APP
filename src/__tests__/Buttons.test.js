import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import Buttons from '../components/Buttons';

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('Button renders  correctly', () => {
  const handleClick = () => {};
  act(() => {
    render(
      <Buttons
        onClick={handleClick}
      />,
      container,
    );
  });

  expect(
    container.querySelector("[class='buttons']").childElementCount,
  ).toEqual(19);
});
