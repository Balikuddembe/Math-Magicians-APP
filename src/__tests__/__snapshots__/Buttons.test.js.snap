// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="buttons"
>
  <button
    type="submit"
  >
    AC
  </button>
  <button
    type="submit"
  >
    +/-
  </button>
  <button
    type="submit"
  >
    %
  </button>
  <button
    className="operator"
    type="submit"
  >
    ÷
  </button>
  <button
    type="submit"
  >
    7
  </button>
  <button
    type="submit"
  >
    8
  </button>
  <button
    type="submit"
  >
    9
  </button>
  <button
    className="operator"
    type="submit"
  >
    ×
  </button>
  <button
    type="submit"
  >
    4
  </button>
  <button
    type="submit"
  >
    5
  </button>
  <button
    type="submit"
  >
    6
  </button>
  <button
    className="operator"
    type="submit"
  >
    -
  </button>
  <button
    type="submit"
  >
    1
  </button>
  <button
    type="submit"
  >
    2
  </button>
  <button
    type="submit"
  >
    3
  </button>
  <button
    className="operator"
    type="submit"
  >
    +
  </button>
  <button
    className="span-two"
    type="submit"
  >
    0
  </button>
  <button
    type="submit"
  >
    .
  </button>
  <button
    className="operator"
    type="submit"
  >
    =
  </button>
</div>
`;

exports[`Button renders  correctly 1`] = `
<div
  className="buttons"
>
   
  <div
    className="buttons"
  >
    <button
      type="submit"
    >
      AC
    </button>
    <button
      type="submit"
    >
      +/-
    </button>
    <button
      type="submit"
    >
      %
    </button>
    <button
      className="operator"
      type="submit"
    >
      ÷
    </button>
    <button
      type="submit"
    >
      7
    </button>
    <button
      type="submit"
    >
      8
    </button>
    <button
      type="submit"
    >
      9
    </button>
    <button
      className="operator"
      type="submit"
    >
      ×
    </button>
    <button
      type="submit"
    >
      4
    </button>
    <button
      type="submit"
    >
      5
    </button>
    <button
      type="submit"
    >
      6
    </button>
    <button
      className="operator"
      type="submit"
    >
      -
    </button>
    <button
      type="submit"
    >
      1
    </button>
    <button
      type="submit"
    >
      2
    </button>
    <button
      type="submit"
    >
      3
    </button>
    <button
      className="operator"
      type="submit"
    >
      +
    </button>
    <button
      className="span-two"
      type="submit"
    >
      0
    </button>
    <button
      type="submit"
    >
      .
    </button>
    <button
      className="operator"
      type="submit"
    >
      =
    </button>
  </div>
</div>
`;

exports[`Calculator renders  correctly 1`] = `
<div
  className="buttons"
>
   
  <div
    className="buttons"
  >
    <button
      type="submit"
    >
      AC
    </button>
    <button
      type="submit"
    >
      +/-
    </button>
    <button
      type="submit"
    >
      %
    </button>
    <button
      className="operator"
      type="submit"
    >
      ÷
    </button>
    <button
      type="submit"
    >
      7
    </button>
    <button
      type="submit"
    >
      8
    </button>
    <button
      type="submit"
    >
      9
    </button>
    <button
      className="operator"
      type="submit"
    >
      ×
    </button>
    <button
      type="submit"
    >
      4
    </button>
    <button
      type="submit"
    >
      5
    </button>
    <button
      type="submit"
    >
      6
    </button>
    <button
      className="operator"
      type="submit"
    >
      -
    </button>
    <button
      type="submit"
    >
      1
    </button>
    <button
      type="submit"
    >
      2
    </button>
    <button
      type="submit"
    >
      3
    </button>
    <button
      className="operator"
      type="submit"
    >
      +
    </button>
    <button
      className="span-two"
      type="submit"
    >
      0
    </button>
    <button
      type="submit"
    >
      .
    </button>
    <button
      className="operator"
      type="submit"
    >
      =
    </button>
  </div>
</div>
`;

exports[`Renders  correctly 1`] = `
<div
  className="buttons"
/>
`;

exports[`Renders  correctly 2`] = `
<div
  className="buttons"
>
   
  <div
    className="buttons"
  >
    <button
      type="submit"
    >
      AC
    </button>
    <button
      type="submit"
    >
      +/-
    </button>
    <button
      type="submit"
    >
      %
    </button>
    <button
      className="operator"
      type="submit"
    >
      ÷
    </button>
    <button
      type="submit"
    >
      7
    </button>
    <button
      type="submit"
    >
      8
    </button>
    <button
      type="submit"
    >
      9
    </button>
    <button
      className="operator"
      type="submit"
    >
      ×
    </button>
    <button
      type="submit"
    >
      4
    </button>
    <button
      type="submit"
    >
      5
    </button>
    <button
      type="submit"
    >
      6
    </button>
    <button
      className="operator"
      type="submit"
    >
      -
    </button>
    <button
      type="submit"
    >
      1
    </button>
    <button
      type="submit"
    >
      2
    </button>
    <button
      type="submit"
    >
      3
    </button>
    <button
      className="operator"
      type="submit"
    >
      +
    </button>
    <button
      className="span-two"
      type="submit"
    >
      0
    </button>
    <button
      type="submit"
    >
      .
    </button>
    <button
      className="operator"
      type="submit"
    >
      =
    </button>
  </div>
</div>
`;
